From 1a9921bab861dd2c89337c3e4833c716a0474653 Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Fri, 12 Feb 2021 03:09:39 +0000
Subject: [PATCH 07/27] uboot-envtools: add defaults for linksys-e8450-ubi

Add U-Boot environment configuration for the Linksys E8450 (UBI) to
allow access to the bootloader environment from OpenWrt via
'fw_printenv' and 'fw_setenv'.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 package/boot/uboot-envtools/files/mediatek | 46 ++++++++++++++++++++++
 1 file changed, 46 insertions(+)
 create mode 100644 package/boot/uboot-envtools/files/mediatek

diff --git a/package/boot/uboot-envtools/files/mediatek b/package/boot/uboot-envtools/files/mediatek
new file mode 100644
index 0000000000..495a837274
--- /dev/null
+++ b/package/boot/uboot-envtools/files/mediatek
@@ -0,0 +1,46 @@
+#
+# Copyright (C) 2021 OpenWrt.org
+#
+
+[ -e /etc/config/ubootenv ] && exit 0
+
+touch /etc/config/ubootenv
+
+. /lib/uboot-envtools.sh
+. /lib/functions.sh
+
+board=$(board_name)
+
+case "$board" in
+linksys,e8450-ubi)
+	ubootenv_add_uci_config "/dev/ubi0_0" "0x0" "0x1f000" "0x1f000" "1"
+	ubootenv_add_uci_config "/dev/ubi0_1" "0x0" "0x1f000" "0x1f000" "1"
+	;;
+bananapi,bpi-r64)
+	. /lib/upgrade/common.sh
+	export_bootdevice
+	export_partdevice rootdev 0
+	case "$rootdev" in
+	mmc*)
+		local envdev=/dev/$(get_partition_by_name $rootdev ubootenv)
+		ubootenv_add_uci_config "$envdev" "0x0" "0x80000" "0x80000" "1"
+		ubootenv_add_uci_config "$envdev" "0x80000" "0x80000" "0x80000" "1"
+		;;
+	*)
+		ubootenv_add_uci_config "/dev/ubi0_0" "0x0" "0x1f000" "0x1f000" "1"
+		ubootenv_add_uci_config "/dev/ubi0_1" "0x0" "0x1f000" "0x1f000" "1"
+		;;
+	esac
+	;;
+buffalo,wsr-2533dhp2)
+	ubootenv_add_uci_config "/dev/mtd3" "0x0" "0x1000" "0x20000"
+	;;
+ubnt,unifi-6-lr-ubootmod)
+	ubootenv_add_uci_config "/dev/mtd2" "0x0" "0x4000" "0x10000"
+	;;
+esac
+
+config_load ubootenv
+config_foreach ubootenv_add_app_config ubootenv
+
+exit 0
-- 
2.25.1

